services/catalog/rs_server_catalog/user_catalog.py

```diff
@@ -89,22 +101,85 @@ def adapt_links(self, content: dict, user: str, collection_id: str, object_name:
             content[object_name][i] = self.adapt_object_links(content[object_name][i], user)
         return content
 
+    @staticmethod
+    def update_stac_item_publication(content: dict, user: str) -> Any:  # pylint: disable=too-many-locals
+        """Update json body of feature push to catalog"""
+        files_s3_key = []
+        # 1 - update assets href
+        for asset in content["assets"]:
+            filename_str = content["assets"][asset]["href"]
+            # Note, conversion to pathlib.Path removes the double / from s3://bucket/path/to/file
+            filename = pathlib.Path(content["assets"][asset]["href"])
+            fid = str(filename).rsplit("/", maxsplit=1)[-1]
+            new_href = (
+                f'https://rs-server/catalog/{user}/collections/{content["collection"]}/items/{fid}/download/{asset}'
+            )
+            content["assets"][asset].update({"href": new_href})
+            # 2 - update alternate href to define catalog s3 bucket
+            s3_key = filename_str.replace(
+                bucket_info["temp-bucket"]["S3_ENDPOINT"],
+                bucket_info["catalog-bucket"]["S3_ENDPOINT"],
+            )
+            new_s3_href = {"s3": {"href": s3_key}}
+            content["assets"][asset].update({"alternate": new_s3_href})
+            files_s3_key.append(filename_str.replace(bucket_info["temp-bucket"]["S3_ENDPOINT"], ""))
+        # 3 - include new stac extension if not present
+
+        new_stac_extension = "https://stac-extensions.github.io/alternate-assets/v1.1.0/schema.json"
+        if new_stac_extension not in content["stac_extensions"]:
+            content["stac_extensions"].append(new_stac_extension)
+        # 4 tdb, bucket movement
+        try:
+            # try with env, but maybe read from json file?
+            handler = S3StorageHandler(
+                os.environ["S3_ACCESSKEY"],
+                os.environ["S3_SECRETKEY"],
+                os.environ["S3_ENDPOINT"],
+                os.environ["S3_REGION"],
+            )
+            config = TransferFromS3ToS3Config(
+                files_s3_key,
+                bucket_info["temp-bucket"]["name"],
+                bucket_info["catalog-bucket"]["name"],
+                max_retries=3,
+            )
```

Comment:
Check if temp-bucket and catalog-bucket exists, otherwise raise a 400 bad request, to prevent features with wrong obs.